src/main/java/it/pagopa/pn/bff/rest/InfoPaController.java

```diff
@@ -0,0 +1,78 @@
+package it.pagopa.pn.bff.rest;
+
+import it.pagopa.pn.bff.exceptions.PnBffException;
+import it.pagopa.pn.bff.generated.openapi.server.v1.api.InstitutionProductsApi;
+import it.pagopa.pn.bff.generated.openapi.server.v1.api.InstitutionsApi;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.BffInstitution;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.BffInstitutionProduct;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.CxTypeAuthFleet;
+import it.pagopa.pn.bff.service.InfoPaService;
+import lombok.CustomLog;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+
+@CustomLog
+@RestController
+public class InfoPaController implements InstitutionProductsApi, InstitutionsApi {
+
+    private final InfoPaService infoPaService;
+
+    public InfoPaController(InfoPaService infoPaService) {
+        this.infoPaService = infoPaService;
+    }
+
+    /**
+     * GET /v1/institutions
+     * Get the list of institutions
+     *
+     * @param xPagopaPnUid          The user id
+     * @param xPagopaPnCxType       The type of the user
+     * @param xPagopaPnCxId         The id of the user
+     * @param xPagopaPnSrcCh        The source channel
+     * @param xPagopaPnCxGroups     The groups of the user
+     * @param xPagopaPnSrcChDetails The details of the source channel
+     * @return The list of institutions
+     */
+    @Override
+    public Mono<ResponseEntity<Flux<BffInstitution>>> getInstitutionsV1(String xPagopaPnUid, CxTypeAuthFleet xPagopaPnCxType, String xPagopaPnCxId, String xPagopaPnSrcCh, List<String> xPagopaPnCxGroups, String xPagopaPnSrcChDetails, ServerWebExchange exchange) {
+        log.logStartingProcess("getInstitutesV1");
+        Flux<BffInstitution> bffInstitutions = infoPaService
+                .getInstitutions(xPagopaPnUid, xPagopaPnCxType, xPagopaPnCxId, xPagopaPnSrcCh, xPagopaPnCxGroups, xPagopaPnSrcChDetails)
+                .onErrorMap(WebClientResponseException.class, PnBffException::wrapException);
+        log.logEndingProcess("getInstitutesV1");
+        return bffInstitutions
+                .collectList()
```

Comment:
maybe collectList is not needed

Reply:
it coverts a stream of elements into a list, i took the inspiration from the real MS: [link](https://github.com/pagopa/pn-external-registries/blob/fce619e3c14a8e57fe8139fb281fa7e04ccc11ea/src/main/java/it/pagopa/pn/external/registries/rest/v1/InfoPaController.java#L112-L117)